src/base/io_buffers.c

```diff
@@ -124,61 +111,31 @@ xdd_init_io_buffers(worker_data_t *wdp) {
 		}
 		if (xgp->global_options & GO_REALLYVERBOSE)
 				fprintf(xgp->output,"Shared Memory ID allocated and attached, shmid=%d\n",buf_shmid);
-#elif (IRIX || WIN32 )
-		fprintf(xgp->errout,"%s: Shared Memory not supported on this OS - using valloc\n",
-			xgp->progname);
-		tdp->td_target_options &= ~TO_SHARED_MEMORY;
-#if (IRIX || SOLARIS || LINUX || AIX || DARWIN || FREEBSD)
 		bufp = valloc(buffer_size);
-#else
-		bufp = malloc(buffer_size);
-#endif
-#endif 
 	} else { /* Allocate memory the normal way */
-#if (AIX || LINUX)
 		posix_memalign((void **)&bufp, sysconf(_SC_PAGESIZE), buffer_size);
-#elif (IRIX || SOLARIS || LINUX || DARWIN || FREEBSD)
```

Comment:
Is this the right choice?

Reply:
It appears so. Interesting that it is always allocated in page sized/aligned chunks. My guess is is this due to this code also handing E2E buffers. See the comment above this function.